src/prism-async.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import highlight from './highlight';
+
+const languages = [];
+
+export const registerLanguage = (_, language) => {
+  if(ReactHighlightAsync.refractor) {
+    return refractor.register(language);
+  } else {
+    languages.push(language);
+  }
+};
+
+export default class ReactHighlightAsync extends React.PureComponent {
+  static refractor = null;
+  static highlightInstance = (highlight(null, {}));
+  static refractorPromise = null;
+  
+  static _loadRefractor() {
+    ReactHighlightAsync.refractorPromise = import(/* webpackChunkName:"react-syntax-highlighter/refractor-import" */ 'refractor/core').then(({ default: refractor }) => {
+      languages.forEach((language) => refractor.register(language));
```

Comment:
this requires every language once it's dynamically imported? I wonder if there's a use case for allowing the specification of a subset of languages? what do you think?

Reply:
Only the languages that had been registered. So the behaviour is the same as the non async loaded variant